src/pages/account/AdminPage.js

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useEffect } from "react";
+import { Link, useParams } from "react-router-dom";
+import styled from "styled-components";
+
+import axiosApi from "../../utils/AxiosApi";
+import Card from "../../components/Card";
+import NavBar from "../../components/NavBar";
+import Button from "../../components/Button";
+
+const AdminPage = () => {
+  const [stories, setStories] = useState([]);
+  const [child, setChild] = useState({});
+  const [id, setId] = useState("");
+  const { childId } = useParams();
+
+  useEffect(() => {
+    const childData = async () => {
+      const response = await axiosApi(`/getChild/${childId}`);
+      const childData = await response.data;
+      const storyData = await response.data.stories;
+      setChild(childData);
+      setStories(storyData);
+    };
+    return childData();
+  }, [childId]);
+
+  const deleteStory = async (event) => {
+    event.preventDefault();
+    if (id) {
+      await axiosApi.post(`deleteStory/${id}`, { childId });
+      const response = await axiosApi(`/getChild/${childId}`);
+      const storyData = await response.data.stories;
+      setStories(storyData);
+    }
+  };
+
+  if (stories.length === 0) {
+    return (
+      <>
+        <NavBar />
+        <p>
+          There are no stories yet. Be the first to write a story and go to your{" "}
+          <Link to="/account">home page</Link>, select {child.firstname} and
+          start writing.
+        </p>
+      </>
+    );
+  }
+
+  return (
+    <>
+      <NavBar />
+      <Header>
+        <h1>{child.firstname}'s Stories</h1>
+      </Header>
+      <Container>
+        {stories.length > 0 &&
+          stories.map((story) => (
+            <Card
+              key={story._id}
+              img={story.media}
+              title={story.title}
+              description={story.description}
+            >
+              <Link to={`/my-stories/details/${story._id}`}>Details</Link>
+              <form onSubmit={deleteStory}>
```

Comment:
This doesn't necessarily need to be in a form.
- You don't have any inputs
- You don't have any form elements

So a form is not necessary here. You can use trigger the function with the button. In that case remove `type=submit`, because `type=submit` is only for the form.


You can pass the id directly to the `deleteStory` function, no need to store it in the `setId` function

Reply:
I think, in that case you can even remove that state

Reply:
If I don't wrap a form around it it doesn't work

Reply:
Well, if you don't call the function in the `onClick` than it doesn't work.
Call the delete function in the `onclick` and pass the story `id` to it

Reply:
<Button onClick={() => deleteStory(storyId)}>Delete</Button> This is what I did right now

Reply:
Yea, that's not entirely it. Search how you can pass data with an `onclick`